File: server/src/main/java/com/logiflow/server/services/user/ProfileService.java
package com.logiflow.server.services.user;

import com.logiflow.server.dtos.user.ProfileDto;

public interface ProfileService {

    ProfileDto getProfile(String username);

    ProfileDto getProfileByUserId(Integer userId);

    ProfileDto updateProfile(String username, com.logiflow.server.dtos.user.ProfileUpdateDto profileUpdateDto);

    void changePassword(String username, String currentPassword, String newPassword);
}

File: client/src/services/profile/profileService.js
import api from '../api';

const changePassword = async (currentPassword, newPassword) => {
  await api.put('/user/profile/password', null, {
    params: { currentPassword, newPassword }
  });
};

const profileService = {
  getProfile: () => api.get('/user/profile').then((res) => res.data),
  updateProfile: (payload) => api.put('/user/profile', payload).then((res) => res.data),
  changePassword,
};

export default profileService;
